File: app/src/main/java/com/projet/sluca/smallbrother/InstallDe2Activity.java
package com.projet.sluca.smallbrother;

import android.content.Context;
import android.content.Intent;
import android.content.pm.PackageManager;
import android.os.Bundle;
import androidx.core.app.ActivityCompat;
import androidx.appcompat.app.AppCompatActivity;
import android.view.Gravity;
import android.view.View;
import android.widget.EditText;
import android.widget.Toast;

public class InstallDe2Activity extends AppCompatActivity
{
    private Context context;                     // Définira l'état courant de l'appli.
    public Vibration vibreur = new Vibration();  // Instanciation d'un vibreur.
    public UserData userdata = new UserData();   // Liaison avec les données globales de l'utilisateur.


    @Override
    protected void onCreate(Bundle savedInstanceState)
    {
        // Etablissement de la liaison avec la vue res/layout/activity_installde.xml.
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_installde2);
        context = this; // Mise à jour de l'état.

        // Etablissement de la liaison avec la classe UserData.
        userdata = (UserData) getApplication();
    }


    // --> Au clic que le bouton "Précédent".
    public void precedent(View view)
    {
        vibreur.vibration(context, 100);

        finish();
    }


    // --> Au clic que le bouton "Terminer".
    public void terminer(View view)
    {
        vibreur.vibration(context, 100);


        // > Récupération du contenu des inputs :

        // Email de l'Aidé :
        EditText et_myemail = (EditText) findViewById(R.id.input_mymail);
        String myemail = et_myemail.getText().toString();

        // Email :
        EditText et_password = (EditText) findViewById(R.id.input_password);
        String password = et_password.getText().toString();


        // > Vérification de la validité des informations entrées :

        // Vérification 1 : l'adresse email n'est pas valide.
        String provider = myemail.substring(myemail.length() - 10);
        if(!myemail.matches("") && !provider.equals("@gmail.com"))// !myemail.contains ("@"))
        {
            message(getString(R.string.error02));
        }
        // Vérification 2 : aucune entrée n'est vide.
        else if(myemail.matches("") || password.matches(""))
        {
            message(getString(R.string.error03));
        }

        // > Validation si tout a été positivement vérifié :

        else
        {
            // Récupération de la version de SB en cours.
            String version = "";
            try { version = context.getPackageManager().getPackageInfo(context.getPackageName(), 0).versionName; }
            catch (PackageManager.NameNotFoundException e) { e.printStackTrace(); }

            // Sauvegarde en globale des valeurs entrées.
            userdata.setRole("Aidé");
            userdata.setMymail(myemail);
            userdata.setPassword(password);

            userdata.saveData(context); // Sauvegarde des données d'utilisateur.

            // Rétablissement des boutons retour, au cas où désactivé par ReglagesActivity.
            userdata.whatAboutGoBack(true);

            // Transition vers l'activity suivante.
            Intent intent = new Intent(context, AideActivity.class);
            startActivityForResult(intent, 1);
        }
    }


    // --> MESSAGE() : affiche en Toast le string entré en paramètre.
    public void message(String message)
    {
        Toast toast = Toast.makeText(getApplicationContext(), message, Toast.LENGTH_LONG);
        toast.setGravity(Gravity.TOP, 0, 0);
        toast.show();

        vibreur.vibration(context, 330);
    }


    // --> Par sécurité : retrait du retour en arrière dans cette activity.
    @Override public void onBackPressed() { moveTaskToBack(false); }
}

File: app/src/main/java/com/projet/sluca/smallbrother/InstallDe2Activity.kt
package com.projet.sluca.smallbrother

import android.content.Intent
import android.content.pm.PackageManager
import android.os.Build
import android.os.Bundle
import android.view.View
import android.widget.EditText
import androidx.activity.OnBackPressedCallback
import androidx.appcompat.app.AppCompatActivity

/***
 * class InstallDe2Activity manages the data of the aide.
 *
 * @author Sébastien Luca & Maxime Caucheteur
 * @version 1.2 (Updated on 10-10-2022)
 */
class InstallDe2Activity : AppCompatActivity() {

    var vibreur = Vibration() // Instanciation d'un vibreur.
    lateinit var userData: UserData // Liaison avec les données globales de l'utilisateur.
    override fun onCreate(savedInstanceState: Bundle?) {
        // Etablissement de la liaison avec la vue res/layout/activity_installde.xml.
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_installde2)

        // Etablissement de la liaison avec la classe UserData.
        userData = application as UserData

        onBackPressedDispatcher.addCallback(this, onBackPressedCallback)
    }

    // --> Au clic que le bouton "Précédent".
    fun precedent(view: View?) {
        vibreur.vibration(this, 100)
        finish()
    }

    // --> Au clic que le bouton "Terminer".
    fun terminer(view: View?) {
        vibreur.vibration(this, 100)


        // > Récupération du contenu des inputs :

        // Email de l'Aidé :
        val etMyEmail = findViewById<EditText>(R.id.input_mymail)
        val myEmail = etMyEmail.text.toString()

        // Email :
        val etPassword = findViewById<EditText>(R.id.input_password)
        val password = etPassword.text.toString()


        // > Vérification de la validité des informations entrées :

        // Vérification 1 : l'adresse email n'est pas valide.
        val provider = myEmail.substring(myEmail.length - 10)
        when {
            !myEmail.matches("".toRegex()) && provider != "@gmail.com"
                -> message(this, getString(R.string.error02), vibreur)

            myEmail.matches("".toRegex()) || password.matches("".toRegex())
                -> message(this, getString(R.string.error03), vibreur)

            else -> {
                // Récupération de la version de SB en cours.
                // (une fois est nécessaire, pas besoin dans InstallDeActivity)
                var version = ""
                try {
                    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.TIRAMISU) {
                        version =
                            this.packageManager
                                .getPackageInfo(
                                    this.packageName,
                                    PackageManager.PackageInfoFlags.of(0)
                                ).versionName
                    } else {
                        @Suppress("DEPRECATION")
                        version =
                            this.packageManager.getPackageInfo(this.packageName, 0).versionName
                    }
                } catch (e: PackageManager.NameNotFoundException) {
                    e.printStackTrace()
                }


                if(intent.hasExtra("nom")){
                    userData.nom = intent.getStringExtra("nom").toString()
                }
                if(intent.hasExtra("mail")){
                    userData.email = intent.getStringExtra("mail").toString()
                }
                if(intent.hasExtra("telephone")){
                    userData.telephone = intent.getStringExtra("telephone").toString()
                }

                // Sauvegarde en globale des valeurs entrées.
                userData.version = version
                userData.role = "Aidé"
                userData.mymail = myEmail
                userData.password = password
                userData.saveData(this) // Sauvegarde des données d'utilisateur.

                // Rétablissement des boutons retour, au cas où désactivé par ReglagesActivity.
                userData.canGoBack = true

                // Transition vers l'activity suivante.
                val intent = Intent(this, AideActivity::class.java)
                startActivity(intent)
            }
        }
    }

    private val onBackPressedCallback = object : OnBackPressedCallback(true) {
        override fun handleOnBackPressed() {
            moveTaskToBack(false)
        }
    }
}